model/schema/core.yaml

```diff
@@ -376,13 +394,65 @@ classes:
 
 slots:
 
-  synonym:
-    description: the text string of the synonym
+  name_type:
+    description: >-
+      The type of name: e.g., symbol, full_name, systematic_name, etc.
+    notes: >-
+      Proposed vocabulary terms for name_type.
+        nomenclature_symbol: A symbol that adheres to nomenclature guidelines: e.g., pax6a, lin-12.
+        full_name: The full length name of an entity: e.g., paired box 6a.
+        systematic_name: A systematic name: e.g., YHR084W, R09F10.2, CG4889.
+        ncbi_protein_name: Synonyms developed by SGD and NCBI curators.
+        uniform: SGD-specific.
+        non_uniform: SGD-specific.
+        retired_name: SGD-specific.
+        unspecified: unclassified names.
+    domain: NameSlotAnnotation
+    range: VocabularyTerm
+    multivalued: false
+    required: true
+
+  name_type_name:
+    description: >-
+      Name of the VocabularyTerm representing the name type of the
+      synonym - proposed values are nomenclature_symbol / full_name /
+      systematic_name / ncbi_protein_name / uniform / non_uniform /
+      retired_name / unspecified
+    range: string
+    required: true
+
+  format_text:
+    description: >-
+      A version of a synonym string using only ASCII characters, which is
+      easier to type (for searches), print and parse. For example, Greek
+      characters are transliterated.
+    aliases: ['synonym_text']
+    domain: NameSlotAnnotation
+    range: string
+    required: true
+    multivalued: false
+
+  display_text:
+    description: >-
+      A version of a synonym string for display. Any UTF8 character is
+      permitted.
+    aliases: ['synonym_sgml']
+    domain: NameSlotAnnotation
```

Comment:
This domain constraint should be removed (or at least relaxed/expanded, if possible) as this slot is also used by the NameSlotAnnotationDTO class

Reply:
Done